dataactvalidator/app.py

```diff
@@ -1,111 +1,73 @@
-        """Handle exceptions explicitly raised during validation."""
-        logger.error(str(error))
-
-        job = get_current_job()
-        if job:
-            if job.filename is not None:
-                # insert file-level error info to the database
-                write_file_error(job.job_id, job.filename, error.errorType, error.extraInfo)
-            if error.errorType != ValidationError.jobError:
-                # job pass prerequisites for validation, but an error
-                # happened somewhere. mark job as 'invalid'
-                mark_job_status(job.job_id, 'invalid')
-        return JsonResponse.error(error, error.status)
-
-    @flask_app.errorhandler(Exception)
-    def handle_validation_exception(error):
-        """Handle uncaught exceptions in validation process."""
-        logger.error(str(error))
-
-        # csv-specific errors get a different job status and response code
-        if isinstance(error, ValueError) or isinstance(error, csv.Error):
-            job_status, response_code = 'invalid', 400
-        else:
-            job_status, response_code = 'failed', 500
-        job = get_current_job()
-        if job:
-            if job.filename is not None:
-                write_file_error(job.job_id, job.filename, ValidationError.unknownError)
-            mark_job_status(job.job_id, job_status)
-        return JsonResponse.error(error, response_code)
-
-    @flask_app.route("/", methods=["GET"])
-    def test_app():
-        """Confirm server running."""
-        return "Validator is running"
-
-    @flask_app.route("/validate/", methods=["POST"])
-    def validate():
-        """Start the validation process."""
-        if request.is_json:
-            g.job_id = request.json.get('job_id')
-        validation_manager = ValidationManager(local, error_report_path)
-        return validation_manager.validate_job(request)
-
-    JsonResponse.debugMode = flask_app.debug
-
-    return flask_app
+    return Flask(__name__)
 
 
 def run_app():
     """Run the application."""
-    flask_app = create_app()
-    flask_app.run(threaded=True, host=CONFIG_SERVICES['validator_host'], port=CONFIG_SERVICES['validator_port'])
+    app = Flask(__name__)
+
+    with app.app_context():
+        current_app.debug = CONFIG_SERVICES['debug']
+        local = CONFIG_BROKER['local']
+        g.is_local = local
+        error_report_path = CONFIG_SERVICES['error_report_path']
+        current_app.config.from_object(__name__)
+
+        # Future: Override config w/ environment variable, if set
+        current_app.config.from_envvar('VALIDATOR_SETTINGS', silent=True)
+
+        @current_app.teardown_appcontext
+        def teardown_appcontext(exception):
+            GlobalDB.close()
+
+        @current_app.before_request
+        def before_request():
+            GlobalDB.db()
+
+        # If local, utilize local db to mock queue
+        if local:
+            sess = GlobalDB.db().session
+
+            while 1:
```

Comment:
I'd bet that hitting the local database like this will have less-than-ideal consequences. What if you added a 10 second sleep (to match the SQS bits below) instead?

Minor: `while True:` is a smidge more Pythonic.